cmake/ecwam_add_test.cmake

```
@@ -18,15 +18,22 @@ endfunction()
 
 function( ecwam_add_test test_name )
 
-    set( one_value_keywords MPI OMP CONFIG )
+    set( one_value_keywords MPI OMP CONFIG PREC)
     set( multi_value_keywords ENVIRONMENT )
 
     include(CMakeParseArguments)
     cmake_parse_arguments( PARSE_ARGV 1 ecwam_add_test "${options}" "${one_value_keywords}" "${multi_value_keywords}" )
 
+    if( ecwam_add_test_PREC )
+      set( test_precision ${ecwam_add_test_PREC} )
+    else()
+      set( test_precision dp )
+    endif()
```

I would not make PREC an optional argument, but rather fail when not provided, to avoid silent bugs.

---

I have a few housekeeping PRs planned, I'll mop this up in one of those. Merging now.